lua/nvim-treesitter/indent.lua

```diff
@@ -128,7 +130,23 @@ function M.get_indent(lnum)
   local node ---@type TSNode
   if is_empty_line then
     local prevlnum = vim.fn.prevnonblank(lnum)
-    node = get_last_node_at_line(root, prevlnum)
+    local indent = vim.fn.indent(prevlnum)
+    local prevline = vim.trim(vim.fn.getline(prevlnum))
+    -- The final position can be trailing spaces, which should not affect indentation
+    node = get_last_node_at_line(root, prevlnum, indent + #prevline - 1)
+    if node:type():match "comment" then
+      -- The final node we capture of the previous line can be a comment node, which should also be ignored
+      -- Unless the last line is an entire line of comment, ignore the comment range and find the last node again
+      local first_node = get_first_node_at_line(root, prevlnum, indent)
+      local srow, scol, erow, _ = node:range()
+      if first_node:id() ~= node:id() and srow == erow then
```

Comment:
I am limiting this to only same line trailing comments as I think there needs to be more discussion of the behavior of multiline comments